plugin-server/src/cdp/hog-transformations/hog-transformer.service.test.ts

```diff
@@ -118,5 +120,142 @@ describe('HogTransformer', () => {
+            })
+
+            mockGetTeamHogFunctions.mockReturnValue([
+                defaultTransformationFunction,
+                geoIpTransformationFunction,
+                testTransformationFunction,
+            ])
+
+            const createHogFunctionInvocationSpy = jest.spyOn(hogTransformer as any, 'createHogFunctionInvocation')
+
+            const event: PluginEvent = {
+                ip: '89.160.20.129',
+                site_url: 'http://localhost',
+                team_id: 1,
+                now: '2024-06-07T12:00:00.000Z',
+                uuid: 'event-id',
+                event: 'event-name',
+                distinct_id: 'distinct-id',
+                properties: { $ip: '89.160.20.129' },
+                timestamp: '2024-01-01T00:00:00Z',
+            }
+
+            await hogTransformer.transformEvent(event)
+
+            expect(createHogFunctionInvocationSpy).toHaveBeenCalledTimes(3)
+            expect(createHogFunctionInvocationSpy).toHaveBeenNthCalledWith(1, event, geoIpTransformationFunction)
+            expect(createHogFunctionInvocationSpy).toHaveBeenNthCalledWith(2, event, defaultTransformationFunction)
+            expect(createHogFunctionInvocationSpy).toHaveBeenNthCalledWith(3, event, testTransformationFunction)
+
+            expect(event.properties?.test_property).toEqual('test_value')
+            expect(event.properties?.$geoip_city_name).toEqual('Linköping')
+        })
+
+        it('should delete a property from previous transformation', async () => {
+            const addingTemplate: HogFunctionTemplate = {
+                status: 'alpha',
+                type: 'transformation',
+                id: 'template-test',
+                name: 'Test Template',
+                description: 'A simple test template that adds a test property',
+                category: ['Custom'],
+                hog: `
+                    let returnEvent := event
+                    returnEvent.properties.test_property := 'test_value'
+                    return returnEvent
+                `,
+                inputs_schema: [],
+            }
+
+            const deletingTemplate: HogFunctionTemplate = {
+                status: 'alpha',
+                type: 'transformation',
+                id: 'template-test',
+                name: 'Test Template',
+                description: 'A simple test template that adds a test property',
+                category: ['Custom'],
+                hog: `
+                    let returnEvent := event
+                    returnEvent.properties.test_property := null
+                    return returnEvent
+                `,
+                inputs_schema: [],
+            }
+
+            const addingTransformationByteCode = await compileHog(addingTemplate.hog)
+            const addingTransformationFunction = createHogFunction({
+                ...addingTemplate,
+                bytecode: addingTransformationByteCode,
+                execution_order: 1,
```

Comment:
Would be cool to try have one that doesn't have an order value (in which case it should go last I would argue)